profiles/common/modules/features/cce_basic_config/cce_basic_config.module

```diff
@@ -341,3 +360,54 @@ function cce_basic_config_modules_enabled($modules) {
     multisite_config_service('user')->grantPermission('editor', array('export nodes'));
   }
 }
+
+/**
+ * Helper function to validate a file's mime type.
+ */
+function _cce_basic_config_validate_real_mimetype(&$errors, $file) {
+  if (variable_get("ec_enable_real_mimetype_validation", TRUE) && !_cce_basic_config_is_cron_run()) {
+    // Get mime type from filename.
+    $mime_by_filename = file_get_mimetype($file->filename);
+    // Get mime type from file contents.
+    $finfo = finfo_open(FILEINFO_MIME_TYPE);
+    // Drupal will in some specific cases append a .txt extension to a file
+    // which will mess with the temporary PHP uri.
+    $temporary_uri = explode(".", $file->uri);
+    $real_mime = finfo_file($finfo, $temporary_uri[0]);
+    if ($mime_by_filename !== $real_mime) {
+      $errors[] = "The file's content type doesn't match the extension.";
```

Comment:
I suggest as message: 

> The file's extension ("[extension]") does not match the file mime type ("[mimetype]")
where:

- [extension] is the extension retrieved from `$file->filename`;
- [mimetype] is the $real_mime value.

We should be as verbose as possible to inform the user about the error.

I think also that we should log the error (watchdog).